### ISSUE TITLE
Sorting on nested relation

### ISSUE THREAD

**Issue by chancezeus** [yajra/laravel-datatables#1977]
### Summary of problem or feature request

When having a model like:

```php
class NestedModel extends Model {
  public function parent()
  {
    return $this->belongsTo(NestedModel::class, 'parent_id');
  }
}
```

If the relation is eager loaded and sorted on, it'll fail with a duplicate table SQL error

### Code snippet of problem

See snippet above

### System details

Latest Laravel Homestead
Latest (released) version of Laravel (5.7.19)
Latest (released) versions of DataTables

### Quickfix
If I place the following code at line 180 of EloquentDataTable.php it'll work (replacing the original performJoin on that line):

```php
            if ($table === $this->query->getModel()->getTable()) {
                $foreign = preg_replace("#^{$table}\\.#", "__nested__{$table}__.", $foreign);
                $deletedAt = $deletedAt ?
                    preg_replace("#^{$table}\\.#", "__nested__{$table}__.", $deletedAt) :
                    $deletedAt;

                $this->performJoin("{$table} as __nested__{$table}__", $foreign, $other, $deletedAt);

                $table = "__nested__{$table}__";
            } else {
                $this->performJoin($table, $foreign, $other, $deletedAt);
            }
```

It's a quick and dirty solution that "works for me", maybe someone can create a more future proof variant, since I didn't check all possible variations (other relation types mostly, this works for BelongsTo, but might fail for the other supported types).

**Comment by yajra** [comment]
Some related links for ref:

https://github.com/yajra/laravel-datatables/issues/880
https://github.com/yajra/laravel-datatables/issues/577
https://github.com/yajra/laravel-datatables/issues/522

**Comment by yajra** [comment]
Fixed via #2051.
